src/sql/workbench/parts/notebook/models/cell.ts

```diff
@@ -187,6 +187,14 @@ export class CellModel implements ICellModel {
 
 	public async runCell(notificationService?: INotificationService, connectionManagementService?: IConnectionManagementService): Promise<boolean> {
 		try {
+			if (!this.active && this !== this.notebookModel.activeCell) {
+				if (this.notebookModel.activeCell) {
```

Comment:
Since I can't remember how we got here, let's just blame me for this design 😛.

Can you file an issue around "We should only track activeCell in one layer"? I know we have this same logic in a number of places, and it would be good as a cleanup item.